sycl/test-e2e/format.py

```diff
@@ -98,6 +98,47 @@ def getMatchedFromList(self, features, alist):
         except ValueError as e:
             raise ValueError("Error in UNSUPPORTED list:\n%s" % str(e))
 
+    def make_default_features_list(self, expr, triple, add_default=True):
+        ### EXCEPTIONS LIST
+        # TODO: Define elsewhere?
+        exceptions = {}
+        exceptions["spir64"]={
+                "cuda":False, "hip":False, "hip_amd":False, "hip_nvidia":False,
+                "native_cpu":False,
+                }
+        exceptions["system"]={
+                "linux":True, "windows":False, "system-windows":False,
+                "run-mode":False, "TEMPORARY_DISABLED":False,
+                }
+        queried_features = []
+        for f in expr:
+            queried_features = queried_features + re.findall("[-+=._a-zA-Z0-9]+", f)
+
+        features = []
+        for f in queried_features:
+            if (exceptions[triple].get(
+                f,exceptions["system"].get(f,add_default))):
+                features.append(f)
+        return features
+
+    def select_triples_for_test(self, test):
+        # Check Triples
+        triples = set()
+        possible_triples = ["spir64"]
+        for triple in possible_triples:
+            unsupported=self.make_default_features_list(test.unsupported,triple,False)
+            required=self.make_default_features_list(test.requires,triple)
+            xfails=self.make_default_features_list(test.xfails,triple,False)
+            if test.getMissingRequiredFeaturesFromList(required):
+                continue
+            if self.getMatchedFromList(unsupported, test.unsupported):
+                continue
```

Comment:
not clear to me how the these functions returning true means we should skip the triple

Reply:
I tried to replicate the functionality of the select_devices_for_test function that already existed in format.py, this bit is from there.

Reply:
ah ok, if it's existing code then nvm thx